blocktracker/blocktracker.go

```diff
@@ -382,31 +370,31 @@ func NewSubscriptionBlockTracker(logger *log.Logger, client *jsonrpc.Client) (*S
 // Track implements the BlockTracker interface
 func (s *SubscriptionBlockTracker) Track(ctx context.Context, handle func(block *ethgo.Block) error) error {
 	data := make(chan []byte)
+	defer close(data)
```

Comment:
Possible race condition here b/c the goroutine that is affected by deferred cancel may still try to write. Can remove this line

Reply:
Have resolved race condition by using context in the callback